src/main/java/hudson/plugins/analysis/core/HealthAwareRecorder.java

```diff
@@ -342,15 +372,14 @@ protected Object readResolve() {
     }
 
     @Override
-    public final boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
```

Comment:
Nop. It's [always returning `true`](https://github.com/jenkinsci/jenkins/blob/9c443c8d5bafd63fce574f6d0cf400cd8fe1f124/core/src/main/java/hudson/tasks/BuildStepCompatibilityLayer.java#L76) when implementing `SimpleBuildStep` (as it's here).